tasks/initialise.yml

```diff
@@ -22,16 +22,35 @@
         PGSETUP_INITDB_OPTIONS: >-
           --encoding=UTF8 --locale=en_US.UTF-8 --auth-host=md5
 
+    - name: postgres | Check for presence of "Modified by ome postgresql ansible role" in file
+      ansible.builtin.lineinfile:
+        path: "{{ postgresql_dist_confdir }}/postgresql.conf"
+        line: "#Modified by ome postgresql ansible role"
+        state: absent
+      check_mode: yes
+      changed_when: false
+      register: config_file_changed
+
+    # read the default postgresql configuration file
+    - name: postgres | get the postgres conf file contents
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.slurp:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf"
+      register: postgres_config_file_contents
+      when: config_file_changed.found | default(0) == 0
+
+
     - name: postgres | postgresql config file
-      template:
+      ansible.builtin.template:
         dest: >-
           {{ postgresql_dist_confdir }}/postgresql.conf
         src: "{{ postgresql_dist_conf_postgresql_src }}"
         mode: 0644
+        owner: "{{ postgresql_become_user }}"
       notify:
         - restart postgresql
+      when: config_file_changed.found | default(0) == 0
 
-      become_user: "{{ postgresql_become_user }}"
```

Comment:
Will this work when the playbook is executed using a privileged non `root` remote user ?